### PR TITLE
[TASK] Add test stage to gitlab CI

### PR DESCRIPTION
**Pull request by lukaszuznanski**
https://gitlab.typo3.org/qa/example-extension/-/pipelines/9553
there is error with job from gitlab template "Code quality" but it's related to free space on machine. Tests are green.

### CODE REVIEW

**Comment by lukaszuznanski** [comment]
https://gitlab.typo3.org/qa/example-extension/-/pipelines/9554

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
     - xmllint --schema ./xliff-core-1.2-strict.xsd
       --noout $(find Resources -name '*.xlf')
 
+unit-tests-php7.2-t3-v10:
+  image: php:7.2
+  stage: test
+  needs:
```
Comment (oliverklee) [inline .gitlab-ci.yml:104]: This also should depend on the PHP 7.2 linting. (If the PHP is not syntactically valid, there is no point in running the test.)

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
...
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+unit-tests-php7.3-t3-v10:
+  image: php:7.3
+  stage: test
+  needs:
```
Comment (oliverklee) [inline .gitlab-ci.yml:112]: This also should depend on the PHP 7.3 linting. (If the PHP is not syntactically valid, there is no point in running the test.)

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
...
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+unit-tests-php7.4-t3-v10:
+  image: php:7.4
+  stage: test
+  needs:
```
Comment (oliverklee) [inline .gitlab-ci.yml:120]: This also should depend on the PHP 7.4 linting. (If the PHP is not syntactically valid, there is no point in running the test.)

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
...
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:unit
+
+functional-tests-php7.2-t3-v10:
+  image: php:7.2
+  services:
+    - mariadb:10
+  stage: test
+  needs:
```
Comment (oliverklee) [inline .gitlab-ci.yml:130]: Like the unit tests, all functional tests should also depend on linting with the corresponding PHP version.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
...
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:functional
+
+functional-tests-php7.3-t3-v10:
```
Comment (oliverklee) [inline .gitlab-ci.yml:135]: Is there a way we could rename the functional and unit testing jobs so their differences are still visible in the pipeline? Currently, it looks like this:
grafik

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -91,6 +98,126 @@ xliff-lint:
...
+  stage: test
+  needs:
+    - build-composer-dependencies
+  script:
+    - composer ci:tests:functional
+
+unit-tests-php7.2-t3-v9:
+  image: php:7.2
+  stage: test
+  dependencies: []
```
Comment (oliverklee) [inline .gitlab-ci.yml:158]: Is this required?

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by lukaszuznanski** [comment]
@oliverklee https://gitlab.typo3.org/qa/example-extension/-/pipelines/9634

File: .gitlab-ci.yml (hunk 6 above)
Comment (lukaszuznanski) [reply .gitlab-ci.yml]: it's required as dependencies are passed from stage to stage and if we have vendor directory, then downgrading to t3 v9 don't work correctly

**Review by oliverklee — APPROVED** [review]

File: .gitlab/build/docker_install.sh [hunk 7]
```diff
@@ -5,9 +5,12 @@
...
 apt-get update -yqq
-apt-get install git libzip-dev unzip libxml2-utils wget -yqq
+apt-get install git libzip-dev unzip parallel libxml2-utils wget -yqq
 
 php -r "readfile('http://getcomposer.org/installer');" | php -- --install-dir=/usr/local/bin/ --filename=composer
 chmod +x /usr/local/bin/composer
 
-docker-php-ext-install zip
+curl --location --output /usr/local/bin/phpunit https://phar.phpunit.de/phpunit.phar
+chmod +x /usr/local/bin/phpunit
```
Comment (oliverklee) [inline .gitlab/build/docker_install.sh:14]: Why are you installing an additional PHPUnit instance? Why not use the one in `vendor/bin/phpunit`?

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by lukaszuznanski** [comment]
@oliverklee https://gitlab.typo3.org/qa/example-extension/-/pipelines/9654

**Review by oliverklee — APPROVED** [review]
Looks good. Could you please rebase to resolve the merge conflicts?

**Review by oliverklee — APPROVED** [review]

**Pull request merged**